numpy/core/src/multiarray/mapping.c

```diff
@@ -1507,7 +1547,7 @@ _get_field_view(PyArrayObject *arr, PyObject *ind, PyArrayObject **view)
                 }
                 Py_DECREF(title);
             }
-            // disallow duplicate field indices
+            /* disallow duplicate field indices */
```

Comment:
Why are we using C-style comments again? If we have a good reason, it seems our tests ought to be catching this...

Reply:
Yes, no test catches this. But it is in the style guide.

We did release 1.13 with these comments present and no-one complained, so maybe it is harmless. We fixed them in the 1.14 branch though.

Also, a grep shows that there is only one other place in numpy with a C++ comment, in `common.c`, which was also added by me in 2016. If we want to remove all C++ comments, I could fix that one into this PR too.

Reply:
We do it because there used to be compilers that didn't accept the C++ style comments. We can probably drop this when we go to c99, although mixed comment styles can be a bit annoying. We should probably test for that, `-ansi` might detect that, although it could give us other errors. IIRC, it didn't work (many years ago).

Reply:
Right, although given that no-one complained in 1.13 it seems no-one building numpy uses incompatible compilers any more.

In any case, I just made the change in `common.c` in this PR, so now numpy is totally free of C++ comments.